[release/2.14] [Bug Skip]: UT failures 2026-6-12

### 🐛 Describe the bug with skip template

Cases:
op_ut,third_party.torch-xpu-ops.test.xpu.test_distributions_xpu.TestDistributionsXPU,test_vonmises_sample_xpu
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOp,test_backward_output_differentiability_tensorlist~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOp,test_backward_partially_registered~~
~~op_ut,third_party.torch-xpu-ops.test.xpu.test_custom_ops_xpu.TestCustomOpAPI,test_no_grad_skips_autograd~~

## ErrorLog
### AssertionError: 0.0188 not less than 0.01859508242727854 : VonMises(loc=0.0, concentration=10.0).sample() is biased:[-0.007  -0.0067 -0.0027 -0.0092 -0.0029  0.0188  0.0077  0.0064 -0.0088
  0.0044]
```bash
________________ TestDistributionsXPU.test_vonmises_sample_xpu _________________
[gw4] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/../../../../test/distributions/test_distributions.py", line 2321, in test_vonmises_sample
    self._check_sampler_sampler(
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/../../../../test/distributions/test_distributions.py", line 1333, in _check_sampler_sampler
    self.assertLess(bias, threshold, message)
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 1232, in assertLess
    self.fail(self._formatMessage(msg, standardMsg))
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 675, in fail
    raise self.failureException(msg)
AssertionError: 0.0188 not less than 0.01859508242727854 : VonMises(loc=0.0, concentration=10.0).sample() is biased:
[-0.007  -0.0067 -0.0027 -0.0092 -0.0029  0.0188  0.0077  0.0064 -0.0088
  0.0044]
```
### AssertionError: False is not true
```bash
________ TestCustomOp.test_backward_output_differentiability_tensorlist ________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/test_custom_ops_xpu.py", line 1588, in test_backward_output_differentiability_tensorlist
    self.assertTrue(c.requires_grad)
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 687, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true
```
### RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```bash
______________ TestCustomOp.test_backward_partially_registered ________________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/test_custom_ops_xpu.py", line 1295, in test_backward_partially_registered
    with self.assertRaisesRegex(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 239, in __exit__
    self._raiseFailure('"{}" does not match "{}"'.format(
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: "unable to find a 'save_for_backward'" does not match "element 0 of tensors does not require grad and does not have a grad_fn"
```
### AssertionError: Scalars are not equal!
```bash
_________________ TestCustomOpAPI.test_no_grad_skips_autograd __________________
[gw7] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
Traceback (most recent call last):
  File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/test_custom_ops_xpu.py", line 2633, in test_no_grad_skips_autograd
    self.assertEqual(called, 1)
  File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 4609, in assertEqual
    raise error_metas.pop()[0].to_error(  # type: ignore[index]
AssertionError: Scalars are not equal!
```
## Pytorch Version
9aa04f02b3aa57a4f34d8d51483ff48b20a5a8a3 

Refer: https://github.com/intel/torch-xpu-ops/actions/runs/27353848409/job/80933328391

### Versions

<details>
<summary>Detail</summary>
Collecting environment information...
PyTorch version: 2.13.0a0+git9aa04f0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar  3 2026, 15:36:40) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-176-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260000
Intel GPU driver version:
* libze1:	1.24.0.0-1146~24.04
* intel-opencl-icd:	25.18.33578.51-1146~24.04
Intel GPU models onboard:
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
* Intel(R) Data Center GPU Max 1100
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-0f00-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [1] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-1600-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [2] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-1a00-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [3] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-1e00-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [4] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-8a00-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [5] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-8e00-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [6] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-c000-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
* [7] _XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1100', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero', type='gpu', device_id=0xBDA, uuid=8680da0b-2f00-0000-c400-000000000000, driver_version='1.6.33578+51', total_memory=49136MB, local_mem_size=128KB, last_level_cache_size=196608KB, max_compute_units=448, memory_clock_rate=3200MHz, memory_bus_width=64-bit, gpu_eu_count=448, gpu_subslice_count=56, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           46 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  192
On-line CPU(s) list:                     0-191
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) Platinum 8468V
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.24.4
[pip3] nvidia-cuda-cupti==13.3.35
[pip3] oneccl==2022.0.0
[pip3] oneccl-devel==2022.0.0
[pip3] onemkl-license==2026.0.0
[pip3] onemkl-sycl-blas==2026.0.0
[pip3] onemkl-sycl-dft==2026.0.0
[pip3] onemkl-sycl-lapack==2026.0.0
[pip3] onemkl-sycl-rng==2026.0.0
[pip3] onemkl-sycl-sparse==2026.0.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2023.0.0
[pip3] tcmlib==1.5.0
[pip3] torch==2.13.0a0+git9aa04f0
[pip3] torchao==0.18.0.dev20260611+xpu
[pip3] torchaudio==2.11.0a0+c0cbdb9
[pip3] torchvision==0.28.0a0+85a1dcc
[pip3] triton-xpu==3.7.2+git5fcc14d9
[pip3] umf==1.1.0
[conda] No relevant packages
</details>

--- a/src/ATen/native/xpu/sycl/DistributionTemplates.h
+++ b/src/ATen/native/xpu/sycl/DistributionTemplates.h
@@ -45,10 +45,12 @@ inline std::tuple<uint64_t, uint32_t, uint32_t> calc_execution_policy(
   auto num_groups = (total_elements + group_size - 1) / group_size;
   auto hw_max_groups = syclMaxWorkItemsPerTile() / group_size;
   num_groups = num_groups > hw_max_groups ? hw_max_groups : num_groups;
-  // number of times random will be generated per thread, to offset philox
-  // counter in thc random state
+  // Increment must be at least the number of 32-bit values consumed per
+  // dist_func call. Our distribution functors consume one rand4-equivalent
+  // block per loop iteration for both float and double paths.
   uint64_t counter_offset =
-      ((total_elements - 1) / (group_size * num_groups * UNROLL) + 1) * UNROLL;
+      ((total_elements - 1) / (group_size * num_groups * UNROLL) + 1) *
+      rand4_engine_calls;
   return std::make_tuple(counter_offset, num_groups, group_size);
 }
 
@@ -133,7 +135,7 @@ void distribution_nullary_kernel(
     return;
   }
 
-  auto execution_policy = calc_execution_policy(numel);
+  auto execution_policy = calc_execution_policy<unroll_factor>(numel);
   auto counter_offset = std::get<0>(execution_policy);
   auto num_groups = std::get<1>(execution_policy);
   auto group_size = std::get<2>(execution_policy);

